distributed/dashboard/components/scheduler.py

```diff
@@ -1720,6 +1724,262 @@ def __del__(self):
 
 
+##########Task Group Graph #########
+class TGroupGraph(DashboardComponent):
+    """
+    Task Group Graph
+    """
+
+    def __init__(self, scheduler, **kwargs):
+        self.scheduler = scheduler
+
+        self.nodes_layout = {}
+        self.arrows_layout = {}
+
+        self.nodes_source = ColumnDataSource(
+            {
+                "x": [],
+                "y": [],
+                "name": [],
+                "name_short": [],
+                "tot_tasks": [],
+                "color": [],
+                "alpha": [],
+            }
+        )
+
+        self.arrows_source = ColumnDataSource({"xs": [], "ys": [], "xe": [], "ye": []})
+
+        self.root = figure(title="Task Groups Graph", **kwargs)
+        self.subtitle = Title(text=" ", text_font_style="italic")
+        self.root.add_layout(self.subtitle, "above")
+
+        w = 5
+        h = 0.5
+        rect = self.root.rect(
+            x="x",
+            y="y",
+            width=w,
+            height=h,
+            color="color",
+            line_color="black",
+            fill_alpha=0.1,
+            source=self.nodes_source,
+        )
+
+        # Example added for update of alpha,
+        w = 4
+        h = 0.25
+
+        rect2 = self.root.rect(
+            x="x",
+            y="y",
+            width=w,
+            height=h,
+            color="color",
+            fill_alpha="alpha",
+            source=self.nodes_source,
+        )
+
+        self.arrows = Arrow(
+            end=VeeHead(size=10),
+            line_color="black",
+            line_alpha=0.5,
+            line_width=1,
+            x_start="xs",
+            y_start="ys",
+            x_end="xe",
+            y_end="ye",
+            source=self.arrows_source,
+        )
+        self.root.add_layout(self.arrows)
+
+        self.labels = LabelSet(
+            x="x",
+            y="y",
+            x_offset=-10,
+            y_offset=20,
+            text="name_short",
+            text_align="left",
+            source=self.nodes_source,
+            background_fill_color=None,
+        )  #  We probably need an offset like x_offset=-1, also need to chop the name somehow
+        self.root.add_layout(self.labels)
+
+        self.root.xgrid.grid_line_color = None
+        self.root.ygrid.grid_line_color = None
+        self.root.x_range.range_padding = 0.5
+        self.root.y_range.range_padding = 0.5
+
+        hover = HoverTool(
+            point_policy="snap_to_data",
+            tooltips=[("tg", "@name"), ("num_task", "@tot_tasks")],
+            renderers=[rect],
+        )
+
+        hover_alpha = HoverTool(
+            point_policy="follow_mouse",
+            tooltips=[("frac", "@alpha")],
+            renderers=[rect2],
+        )
+
+        self.root.add_tools(hover)
+        self.root.add_tools(hover_alpha)
+
+    @without_property_validation
+    def update_layout(self):
+
+        with log_errors():
+            if self.scheduler.task_groups.keys():
```

Comment:
Currently, `update_layout` will return `None` when there are no task groups on the scheduler which is leading to the
 
```python
E           TypeError: cannot unpack non-iterable NoneType object
```

error we're seeing in tests right now. I think we can remove this `if self.scheduler.task_groups.keys():` check and indent the rest of this method 4 spaces to the left and everything will "just work".